pkg/manifests/external_dns.go

```diff
@@ -54,41 +86,45 @@ func (p Provider) String() string {
 	}
 }
 
-func (p Provider) ResourceName() string {
-	switch p {
-	case PublicProvider:
-		return externalDnsResourceName
-	case PrivateProvider:
-		return externalDnsResourceName + "-private"
-	default:
-		return ""
+// ExternalDnsConfig defines configuration options for required resources for external dns
+type ExternalDnsConfig struct {
+	TenantId, Subscription, ResourceGroup,
+	ClientId, ServiceAccountName, Namespace,
+	CRDName string
+	IdentityType       IdentityType
+	ResourceTypes      []ResourceType
```

Comment:
seems fine but would be better to make this a set to avoid any potential for misuse